src/realtime.js

```diff
@@ -719,6 +719,16 @@
      */
 
     on: function (name, handler, context, once) {
+      if(name.startsWith('socket:')){
```

Comment:
I don't really understand here what it is done and why it could not be mutualised with code below. Why don't you just test if `name.startsWith('socket:')` and then alter `name` like `name = name.split('socket:')[1]` and keep the rest of the logic?

Reply:
Because we still need socket.io to send us notifications when we get a message in the websocket. If we don't add the socket.on it won't know we need to listen to additional messages. It's different with the xhr because we're the ones triggering the event when pulling events.

Reply:
Ok I understand. Some thoughts about that:

- Currently, we opted for a logic where this lib "controls" which events could be listened. We could say that to @krichprollsch and ask him for a generic event name like `whosin` which could have that kind of payload: 

```
{ type: 'join|leave', id: 'hash', room: 'name' } 
```

We'll then only have to register a new `.on(` similar to broadcast in `bindSocketEvents`

- If we want to be able to listen to any custom event name, we could rather work in `bindSocketEvents` too [(maybe with middleware?)](https://socket.io/docs/server-api/#socket-use-fn) to be able to catch these and again bind them to `.on()`

My point here is if feel it wrong to re-define the calling of the handle and even the `once` method in `on()` since this is already done in the vendor (with `trigger`)

WDYT?